### PR TITLE
Clear updates

### PR DESCRIPTION
**Pull request by BriungRi**
- Allow `router_signer` to do privileged actions in v1
- Delete clear_registration
- Add test for repeated registration/clear
- Have router use `force_clear_reigstration` instead of `clear_registration
  - Clear registration upon migration and registration

### CODE REVIEW

**Comment by angieyth** [comment]
if we need to use `force_` functions, should we call those from router, or should we move those functions to router?

File: router/sources/router.move [hunk 1]
```diff
@@ -190,6 +190,8 @@ module router::router {
                 domain_name,
                 registration_duration_secs,
             );
+            // Clear the name in v1
+            domains::force_clear_registration(&get_router_signer(), option::none(), domain_name)
```
Comment (rtso) [inline router/sources/router.move:194]: Are we also handling clearing the v1 primary name somewhere? I'd suggest handling it inside of `force_clear_registration`.

File: router/sources/router.move [hunk 2]
```diff
@@ -361,8 +364,10 @@ module router::router {
             );
             let token_id = aptos_names::token_helper::latest_token_id(&tokendata_id);
 
-            // Clear the registration in v1. This will also clear the primary name if applicable
-            domains::clear_registration(user, subdomain_name, domain_name);
+            // Clear the primary name in v1
+            if (is_primary_name) {
+                domains::clear_reverse_lookup(user);
+            };
```
Comment (rtso) [inline router/sources/router.move:370]: This could be cleaner if we handled it within `force_clear_registration`

**Comment by rtso** [comment]
LGTM, I'll defer to @angieyth and @0xaptosj for the final stamp.

**Review by angieyth — APPROVED** [review]
looking good

**Review by 0xaptosj — APPROVED** [review]

**Pull request merged**